examples/hnas/abohb_example.py

```diff
@@ -0,0 +1,229 @@
+    parser.add_argument('--hostfile', type=argparse.FileType('r'))
+    parser.add_argument('--store_results_period', type=int, default=100,
+                        help='If specified, results are stored in intervals of '
+                             'this many seconds (they are always stored at '
+                             'the end)')
+    parser.add_argument('--scheduler', type=str, default='hyperband_promotion',
+                        choices=['hyperband_stopping', 'hyperband_promotion'],
+                        help='Asynchronous scheduler type. In case of doubt leave it to the default')
+    args = parser.parse_args()
+    return args
+
+
+if __name__ == "__main__":
+
+    args = parse_args()
+
+    logging.root.setLevel(logging.INFO)
+
+    # In case you want to run AutoGluon across multiple instances, you need to provide it with a list of
+    # the IP addresses of the instances. Here, we assume that the IP addresses are stored in a .yaml file.
+    # If you want to run AutoGluon on a single instance just pass None.
+    # However, keep in mind that it will still parallelize the optimization process across multiple threads then.
+    # If you really want to run it purely sequentially, set the num_cpus equal to the number of VCPUs of the machine.
+    dist_ip_addrs = yaml.load(args.hostfile) if args.hostfile is not None else []
+
+    del args.hostfile
+
+    print("Got worker host IP addresses [{}]".format(dist_ip_addrs))
+
+    if args.scheduler == "hyperband_stopping":
+        hyperband_type = "stopping"
+
+    elif args.scheduler == "hyperband_promotion":
+        hyperband_type = "promotion"
+
+    brackets = 1  # setting the number of brackets to 1 means that we run effectively successive halving
+
+    scheduler = ag.scheduler.HyperbandScheduler(objective_function,
+                                                resource={'num_cpus': args.num_cpus, 'num_gpus': args.num_gpus},
+                                                # Autogluon runs until it either reaches num_trials or time_out
+                                                num_trials=args.num_trials,
+                                                time_out=args.timeout,
+                                                # This argument defines the metric that will be maximized.
+                                                # Make sure that you report this back in the objective function.
+                                                reward_attr='performance',
+                                                # The metric along we make scheduling decision. Needs to be also
+                                                # reported back to AutoGluon in the objective function.
+                                                time_attr='epoch',
+                                                brackets=brackets,
+                                                max_t=27,  # maximum number of epochs to train the network
```

Comment:
You do not need to set this, it will get it from the train_fn (from args.epochs).